src/v/cluster/id_allocator_stm.h

```diff
@@ -34,10 +34,7 @@ class id_allocator_stm final : public raft::persisted_stm<> {
 public:
     static constexpr std::string_view name = "id_allocator_stm";
 
-    struct stm_allocation_result {
-        int64_t id;
-        raft::errc raft_status{raft::errc::success};
-    };
+    using stm_allocation_result = result<int64_t, raft::errc>;
```

Comment:
i see that this is semantically the same as the previous type but i think we can use simple `result<int64_t>` here. We do not need to specify a category here ?

Reply:
`raft_fixture::retry_with_leader` wants the callback result type to be `raft::errc`-assignable:
https://github.com/redpanda-data/redpanda/blob/eb003fe28cc87b99810d8ea94183a2dbe89399c2/src/v/raft/tests/raft_fixture.h#L409

Reply:
I'll try to make `raft_fixture::retry_with_leader` support `raft::errc`, `cluster::errc` and `std::error_code`